src/DataFramesMeta.jl

```diff
@@ -44,6 +44,178 @@ replace_syms!(e::Expr, membernames) =
+
+    
+`@row` transforms an expression of the form `z = :x + :y` into it's equivalent in 
+DataFrames's "mini-language". Functions act row-wise. For column-wise functions, see 
+`@col`
+
+### Details
+
+Parsing follows the same convention as other DataFramesMeta macros, such as `@with`. All 
+terms in the expression that are `Symbols` are treated as columns in the DataFrame, except 
+`Symbol`s wrapped in `^`. To use a variable representing a column name, wrap the variable 
+in `cols`. 
+
+`@row` constructs an anonymous function based off the given expression. It then creates
+a `source => fun => destination` pair that is suitable for the `select`, `transform`, and 
+`combine` functions in DataFrames. 
+
+### Examples
+
+```
+julia> @row z = :x + :y
+[:x, :y] => (ByRow{var"###607"}(##607) => :z)
+
+```
+
+In the above example, `##607` is an anonymous function equivelent to the following 
+
+```
+(_x, _y) -> _x + _y
+```
+
+The function is wrapped in `ByRow` indicating that DataFrames applies the function 
+to each row of the data frame.  
+
+```
+julia> df = DataFrame(x = [1, 2], y = [3, 4]);
+
+julia> DataFrames.transform(df, @row z = :x * :y)
+2×3 DataFrame
+│ Row │ x     │ y     │ z     │
+│     │ Int64 │ Int64 │ Int64 │
+├─────┼───────┼───────┼───────┤
+│ 1   │ 1     │ 3     │ 3     │
+│ 2   │ 2     │ 4     │ 8     │
+
+julia> DataFrames.transform(df, [:x, :y] => ByRow((_x, _y) -> _x * _y) => :z)
+2×3 DataFrame
+│ Row │ x     │ y     │ z     │
+│     │ Int64 │ Int64 │ Int64 │
+├─────┼───────┼───────┼───────┤
+│ 1   │ 1     │ 3     │ 3     │
+│ 2   │ 2     │ 4     │ 8     │
+"""
+macro row(kw)
+    esc(fun_to_vec(kw; byrow = true))
+end
+
+
+function fun_to_vec(kw::Expr; byrow = false)
+    # z = :x + :y
+    if kw.head == :(=) || kw.head == :kw
+        # New column to be created
+        output = kw.args[1]
+
+        # membernames:
+        # Dict(:x => _x, :y => _y)
+        #
+        # body:
+        # :(function(_x, _y) _x + _y end) 
+        membernames = Dict{Any, Symbol}()
+        funname = gensym()
+        body = replace_syms!(kw.args[2], membernames)
+
+        if byrow == false
+            # z = :x + :y
+            if kw.args[1] isa Symbol
+                t = quote
+                    $(Expr(:vect, keys(membernames)...)) => function $funname($(values(membernames)...))
+                        $body 
+                    end => $(QuoteNode(output))
+                end
+            # n = :z
+            # cols(n) = :x + :y
+            elseif DataFramesMeta.onearg(kw.args[1], :cols)
+                t = quote
+                    $(Expr(:vect, keys(membernames)...)) => function $funname($(values(membernames)...))
+                        $body 
+                    end => $(output.args[2])
+                end   
+            end
+        else 
+            if kw.args[1] isa Symbol
+                t = quote
+                    $(Expr(:vect, keys(membernames)...)) => $(ByRow)(function $funname($(values(membernames)...))
```

Comment:
Since everything is the same except `ByRow`, would it be possible to do something like `identity` when `!byrow` so that code can be shared?

Reply:
> Cool. Though maybe these shouldn't be exported until we have a clear idea of the API we want to provide? IIUC all macros will apply this automatically, right?

Yes, `@transform` will just be `transform(df, @col ...)` etc. 

I would also probably add `@combine` and `@filter` eventually so there is a 1:1 parallel. 


> Yeah. `@bycol` and `@byrow` would make sense to parallel `ByRow` in DataFrames, but we already have `@byrow`, so...

We should rename `@byrow` to `@eachrow` imo. Since it's use and syntax follows super closely with `for row in eachrow(df)...`

> Would it still be possible to support `All`, `Between`, `Not` when they are used alone?

Yes, it would be possible. However this PR has the ability to be a totally non-breaking, no behavior-changing change to DataFramesMeta. So I guess I would rather merge this without that feature and ensure that everything is the same and then add this behavior in a new PR.